hyperjumptech/monika
# Fix [552]:  memory leak on symon mode

# Monika Pull Request (PR)  

## What feature/issue does this PR add  
There are memory leaks when monika is run in symon-mode and no connection to the symon

## How did you implement / how did you fix it  
1. do not send report if there is no connection to symon 
2. clear interval for reporting to symon when there is no connection
3. set new interval for sending report when symon is re connected

## How to test  
1. install node clinic js : `npm i -g clinic`
2. run monika in symon mode using clinic : `clinic doctor -- node bin/run --symonUrl https://symon.hyperjump.tech --symonKey YOUR_KEY_HERE --verbose`
3. let monika run for 2 minutes
4. kill connection
5. let monika run for another 3 minutes
6. stop monika by using `ctrl + c`, wait untill the clinic js result is showing on your default browser 
7. It should show something like in the below.

![Screen Shot 2022-01-13 at 16 40 13](https://user-images.githubusercontent.com/8952493/149305153-bd241ac6-996e-40db-a631-db1cdfee53f7.png)
